flink-connectors/flink-connector-jdbc/src/test/java/org/apache/flink/connector/jdbc/xa/SemanticXidGeneratorTest.java

```diff
@@ -40,21 +40,25 @@ public void testXidsUniqueAmongCheckpoints() {
     }
 
     @Test
-    public void testXidsUniqueAmongGenerators() {
+    public void testXidsUniqueAmongJobs() {
         long checkpointId = 1L;
+        SemanticXidGenerator generator = new SemanticXidGenerator();
         checkUniqueness(
                 unused -> {
-                    SemanticXidGenerator generator = new SemanticXidGenerator();
                     generator.open();
                     return generator.generateXid(TEST_RUNTIME_CONTEXT, checkpointId);
                 });
     }
 
     private void checkUniqueness(Function<Integer, Xid> generate) {
         Set<Xid> generated = new HashSet<>();
+        Set<byte[]> gtrids = new HashSet<>();
```

Comment:
I'm afraid `byte[]` can't be used as a hashtable key because it's equals/hashCode doesn't take array contents into accout.

Reply:
You're right, fixed